src/Bicep.Core/Analyzers/Linter/Rules/NoUnusedVariablesRule.cs

```diff
@@ -29,14 +30,13 @@ public override string FormatMessage(params object[] values)
 
         override public IEnumerable<IDiagnostic> AnalyzeInternal(SemanticModel model, DiagnosticLevel diagnosticLevel)
         {
-            // TODO: Performance: Use a visitor to visit VariableAccesssyntax and collects the non-error symbols into a list.
-            // Then do a symbol visitor to go through all the symbols that exist and compare.
-            // Same issue for unused-params rule.
+            var invertedBindings = model.Binder.Bindings.ToLookup(kvp => kvp.Value, kvp => kvp.Key);
```

Comment:
[nit]: Since all the `NoUnused*Rule`s use the same binding inversion, would it make sense to just expose this as a property of the binder? It looks like a cheap thing to calculate, but it could be done once and cached as an `ImmutableDictionary<Symbol, ImmutableArray<SyntaxBase>>`.